src/main/java/net/logstash/logback/composite/FormattedTimestampJsonProvider.java

```diff
@@ -160,11 +96,11 @@ protected String getFormattedTimestamp(Event event) {
      */
     private void updateTimestampWriter() {
         if (UNIX_TIMESTAMP_AS_NUMBER.equals(pattern)) {
-            timestampWriter = new NumberTimestampWriter();
+            timestampWriter = null;
```

Comment:
This is a backwards incompatible change that I won't be able to accept.

logstash-logback-encoder still needs to optionally support writing timestamps as numbers indicating the number of milliseconds since unix epoch

Reply:
For this one, you have to take a close look at the overall change, or maybe I should remove UNIX_TIMESTAMP_AS_NUMBER completely because currently its only used in 2 test cases for it....
However, what was done previously, you would take the timestamp as milliseconds, convert them to a Date, and then take this NumberTimestampWriter to write it back as timestamp in milliseconds.

So actually im not using it anymore, I will remove it to clear confusion.
(Nothing lost here)

Reply:
We cannot remove the UNIX_TIMESTAMP_AS_NUMBER, or the `NumberTimestampWriter`.

The `FormattedTimestampJsonProvider` supports three scenarios:

UNIX_TIMESTAMP_AS_NUMBER will write the timestamp value as a json _number_ indicating the number of milliseconds since unix epoch, like this:

```
{
    "@timestamp" : 1570901563
}
```

UNIX_TIMESTAMP_AS_STRING will write the timestamp value as a json _string_ indicating the number of milliseconds since unix epoch, like this:

```
{
    "@timestamp" : "1570901563"
}
```

And any other pattern will be used as a format string.  e.g. 

```
{
    "@timestamp" : "2011-12-03T10:15:30+01:00"
}
```

From what I can tell, the new code always writes the timestamp value as a json _string_, which has effectively removed support for UNIX_TIMESTAMP_AS_NUMBER.